cmd/nerdctl/rename.go

```diff
@@ -0,0 +1,168 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/nerdctl/pkg/dnsutil/hostsstore"
+	"github.com/containerd/nerdctl/pkg/idutil/containerwalker"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+const (
+	// hostsDirBasename is the base name of /var/lib/nerdctl/<ADDRHASH>/etchosts
+	hostsDirBasename = "etchosts"
+	// metaJSON is stored as /var/lib/nerdctl/<ADDRHASH>/etchosts/<NS>/<ID>/meta.json
+	metaJSON = "meta.json"
+)
+
+var dataStore string
+var ns string
+
+func newRenameCommand() *cobra.Command {
+	var renameCommand = &cobra.Command{
+		Use:               "rename [flags] CONTAINER ",
+		Args:              cobra.MinimumNArgs(2),
+		Short:             "Rename a container",
+		RunE:              renameAction,
+		ValidArgsFunction: renameShellComplete,
+		SilenceUsage:      true,
+		SilenceErrors:     true,
+	}
+	return renameCommand
+}
+
+func renameAction(cmd *cobra.Command, args []string) error {
+	if len(args) != 2 {
```

Comment:
not executed when setting this https://github.com/containerd/nerdctl/pull/610/files#diff-835246645110170f61bdc679ba318365c436b374e5c1befc2c2200e8de76dc7aR47